### PR TITLE
fix(meter): fix infinite loop caused by loop variable type mismatch

### PR DESCRIPTION
**Pull request by FASTSHIFT**
### Checkpoints
- [x] Follow the [styling guide](https://github.com/lvgl/lvgl/blob/master/docs/CODING_STYLE.md)
- [x] Run `code-format.py` from the `scripts` folder. [astyle](http://astyle.sourceforge.net/install.html) needs to be installed.
- [ ] Update the documentation

### CODE REVIEW

File: src/extra/widgets/meter/lv_meter.c [hunk 1]
```diff
@@ -435,7 +433,7 @@ static void draw_ticks_and_labels(lv_obj_t * obj, lv_draw_ctx_t * draw_ctx, cons
         int16_t inner_act_mask_id = LV_MASK_ID_INV; /*Will be added later*/
 
         uint32_t minor_cnt = scale->tick_major_nth ? scale->tick_major_nth - 1 : 0xFFFF;
-        for(i = 0; i < scale->tick_cnt; i++) {
+        for(uint16_t i = 0; i < scale->tick_cnt; i++) {
```
Comment (kisvegabor) [inline src/extra/widgets/meter/lv_meter.c]:
```suggestion
        uint16_t i;
        for(i = 0; i < scale->tick_cnt; i++) {
```

**Comment by kisvegabor** [comment]
Thank you for this and all the other fixes. Have you found these with a static code analyzer?

**Comment by FASTSHIFT** [comment]
> Thank you for this and all the other fixes. Have you found these with a static code analyzer?

Yes，I used [Coverity](https://scan.coverity.com) and [TscanCode](https://github.com/Tencent/TscanCode).

**Comment by kisvegabor** [comment]
AFAIK, Coverity can run only online, but seemingly TscanCode can run offline too. If so it'd be great to have a script that runs TscanCode. This way we can run it at least manually time to time.

**Comment by FASTSHIFT** [comment]
> AFAIK, Coverity can run only online, but seemingly TscanCode can run offline too. If so it'd be great to have a script that runs TscanCode. This way we can run it at least manually time to time.

TscanCode has a [linux command line tool](https://github.com/Tencent/TscanCode/tree/master/release/linux/TscanCodeV2.14.2395.linux), which can be called as a shell script.
Consider integrating into CI.

**Comment by kisvegabor** [comment]
I've just tested with `./tscancode src/. 2>result.txt` and it reported 126 problem.

If we integrated it into CI we needed to fix these 126 problems first. :slightly_smiling_face:

**Pull request merged**